index.js

```diff
@@ -0,0 +1,149 @@
+const compareLangKeys = () => {
+  // lib is needed to show deuplicated keys
+  const jsonValidator = require('json-dup-key-validator');
+
+  //requiring path and fs modules
+  const path = require('path');
+  const fs = require('fs');
+  const directoryPath = path.join(__dirname, 'locales');
+
+  const mapAllFiles = (files, filesContent) => {
+    const isJsonFile = (JSONdata, file) => {
+      const singleFilesContent = [];
+
+      for (const key in JSONdata) {
+        singleFilesContent.push(key);
+      }
+      filesContent.push({
+        fileName: file,
+        fileKeys: singleFilesContent
+      });
+    };
+
+    const isNotJsonFile = () => {
+      notJSON++;
+      filesContent.push({
+        fileName: 'null',
+        fileKeys: []
+      });
+    };
+
+    const loadFiles = (file, isJsonFile, isNotJsonFile) => {
+      // load file if it is a JSON
+      if (file.includes('.json')) {
+        const JSONdata = require(`./locales/${file}`);
+
+        const isDuplicated = new Promise((resolve, reject) => {
+          fs.readFile(`./locales/${file}`, 'utf8', (err, data) => {
+            if (err) {
+              reject(err);
+            } else {
+              const validateDuplicates = jsonValidator.validate(data, false);
+              if (validateDuplicates) {
+                resolve(
+                  `[DUPLICATED] ${jsonValidator.validate(
+                    data,
+                    false
+                  )} => in ${file}`
+                );
+              }
+            }
+          });
+        });
+        isDuplicated.then(res => {
+          console.info(res);
+        });
+        isDuplicated.catch(rej => {
+          console.info(rej);
+        });
+
+        isJsonFile(JSONdata, file);
+      } else {
+        isNotJsonFile;
+      }
+    };
+
+    files.forEach(file => {
+      loadFiles(file, isJsonFile, isNotJsonFile);
+    });
+  };
+  const countKeysBetweenFiles = filesContent => {
+    const singleFinalKeysArr = [];
```

Comment:
A [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) would be helpful here:

```suggestion
    const keys = filesContent.flatMap(file => file.fileKeys);
```